trustchain-http/src/verifier.rs

```diff
@@ -1,94 +1,400 @@
+use crate::config::http_config;
+use crate::errors::TrustchainHTTPError;
 use crate::qrcode::str_to_qr_code_html;
 use crate::state::AppState;
-use crate::EXAMPLE_VP_REQUEST;
-use axum::extract::State;
+use async_trait::async_trait;
+use axum::extract::{Path, State};
 use axum::http::StatusCode;
 use axum::response::{Html, IntoResponse};
 use axum::Json;
 use log::info;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use ssi::did_resolve::DIDResolver;
+use ssi::ldp::LinkedDataDocument;
 use ssi::vc::{Credential, Presentation};
 use std::sync::Arc;
+use trustchain_api::api::TrustchainVPAPI;
+use trustchain_api::TrustchainAPI;
+use trustchain_core::verifier::{Timestamp, Verifier};
+use trustchain_ion::verifier::IONVerifier;
 
-// TODO: implement in core?
-pub struct PresentationRequest;
-
-// TODO: implement in core?
-/// An error type for presentation failures
-pub enum PresentationError {
-    FailedToVerify,
-    // TODO: add other variants
-}
+/// A type for presentation requests. See [VP request spec](https://w3c-ccg.github.io/vp-request-spec/)
+/// for further details.
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct PresentationRequest(Value);
 
 /// An API for a Trustchain verifier server.
+#[async_trait]
 pub trait TrustchainVerifierHTTP {
-    /// Constructs a presentation request (given some `presentiation_id`) to send to a credential holder from request wallet by ID
-    fn generate_presentation_request(presentation_id: &str) -> PresentationRequest;
-    /// Verifies verifiable presentation
-    fn verify_presentation(presentation: &Presentation) -> Result<(), PresentationError>;
-    /// Verifies verifiable credential
-    fn verify_credential(credential: &Credential) -> Result<(), PresentationError>;
+    /// Constructs a presentation request (given some `presentiation_id`) to send to a credential
+    /// holder from request wallet by ID.
+    fn generate_presentation_request(_presentation_id: &str) -> PresentationRequest {
```

Comment:
Remove method?

Reply:
Can be left as a TODO, I think.

Reply:
Removing as request generated directly in handler.